### PR TITLE
Configurable Inbound/Outbound HTTP/1 Connection Idle Timeouts

### PR DESCRIPTION
**Pull request by txabman42**
Set `LINKERD2_PROXY_INBOUND_HTTP1_CONNECTION_POOL_IDLE_TIMEOUT` and `LINKERD2_PROXY_OUTBOUND_HTTP1_CONNECTION_POOL_IDLE_TIMEOUT` during injection

This MR introduces the ability to configure inbound and outbound HTTP/1 connection pool idle timeouts on a per-service. Currently, linkerd2-proxy has default idle timeouts of 3 seconds for both inbound and outbound HTTP/1 connections.

For example, this change is beneficial for services requiring persistent TCP connections across operational rounds.

There was the intention to do something similar, as commented some time ago here https://github.com/linkerd/linkerd2/issues/9273#issuecomment-1241116227:
> **Proposal**
> `2.13.0`
> Make this timeout configurable via the policy controller, controlled by an annotation set on a Server or Pod.

This MR aims to add the corresponding annotations in order to make it configurable via annotations in the pod.

### CODE REVIEW

**Comment by adleong** [comment]
Thanks @txabman42!  Structurally, this change looks good but I want to make sure this won't conflict with future plans to make this configurable through the proxy API (as indicated by https://github.com/linkerd/linkerd2-proxy/blob/main/linkerd/app/src/env.rs#L286-L294).  @olix0r do you have any thoughts on how near-term making this configurable through the API is?

**Comment by txabman42** [comment]
@olix0r could you take and eye to this MR?

File: charts/linkerd-control-plane/templates/identity.yaml [hunk 1]
```diff
@@ -215,6 +215,8 @@ spec:
       {{- $_ := set $tree.Values.proxy "capabilities" (dict "drop" (list "ALL")) }}
       {{- $_ := set $tree.Values.proxy "outboundDiscoveryCacheUnusedTimeout" "5s" }}
       {{- $_ := set $tree.Values.proxy "inboundDiscoveryCacheUnusedTimeout" "90s" }}
+      {{- $_ := set $tree.Values.proxy "inboundHTTP1ConnectionPoolIdleTimeout" "3s" }}
+      {{- $_ := set $tree.Values.proxy "outBoundHTTP1ConnectionPoolIdleTimeout" "3s" }}
```
Comment (mateiidavid) [inline charts/linkerd-control-plane/templates/identity.yaml:219]: This is a bit nitpicky but we generally don't capitalise the B in outbound.

```suggestion
      {{- $_ := set $tree.Values.proxy "outboundHTTP1ConnectionPoolIdleTimeout" "3s" }}
```

**Review by mateiidavid — COMMENTED** [review]
Thanks for opening this :) It seems like we do not have any entries in the Helm values file. We should add it in. Also, can we please add the annotations to `cmd/cli/doc.go`?

File: charts/linkerd-control-plane/templates/destination.yaml [hunk 2]
```diff
@@ -184,6 +184,8 @@ spec:
       {{- $_ := set $tree.Values.proxy "podInboundPorts" "8086,8090,8443,9443,9990,9996,9997" }}
       {{- $_ := set $tree.Values.proxy "outboundDiscoveryCacheUnusedTimeout" "5s" }}
       {{- $_ := set $tree.Values.proxy "inboundDiscoveryCacheUnusedTimeout" "90s" }}
+      {{- $_ := set $tree.Values.proxy "inboundHTTP1ConnectionPoolIdleTimeout" "3s" }}
+      {{- $_ := set $tree.Values.proxy "outBoundHTTP1ConnectionPoolIdleTimeout" "3s" }}
```
[the remark this answers is not in the document]
Comment (txabman42) [reply charts/linkerd-control-plane/templates/destination.yaml]: Good catch, updated

File: charts/linkerd-control-plane/templates/identity.yaml (hunk 1 above)
Comment (txabman42) [reply charts/linkerd-control-plane/templates/identity.yaml]: Good catch, updated

Comment (txabman42) [reply charts/linkerd-control-plane/templates/identity.yaml]: `cli/cmd/doc.go` has been also updated.

**Comment by txabman42** [comment]
@mateiidavid could you review it again? TY in advance

**Comment by kflynn** [comment]
@txabman42 Happy new year! Think you can resolve the conflict and take a look at the failing test here? 🙂

**Comment by txabman42** [comment]
@kflynn Conflict resolved and tests fixed, and happy new year!!!

**Comment by mateiidavid** [comment]
Hi, @txabman42! Thanks for your patience. It's taken us quite a bit of time to review this since we've been focused on upcoming releases. The change itself looks good, but I think we'd like to have a bit more time to think through the contract that we're introducing here. This includes the annotation names, the annotation key domain, and whether this should be configured globally, per-workload, or both.

In the mean time, since this clearly addresses a problem you're having, I wanted to suggest a stop gap solution. We've introduced a feature that allows us to use static configuration overrides (through environment variables). These would apply globally; you can use them to tweak the idle timeouts.

It's not documented but you can have a look at:
* [The PR that introduced them](https://github.com/linkerd/linkerd2/pull/11874)
* [The latest PR that changed the Helm value format](https://github.com/linkerd/linkerd2/pull/11923)

**Comment by mateiidavid** [comment]
@txabman42 going to close this PR since it's been open for a long time. Please let me know if the workaround I mentioned above is good for you. If you want this re-open, let us know. :)

**Pull request closed without merging**